tensorflow/compiler/tf2tensorrt/utils/trt_tensor_proxy.h

```diff
@@ -143,28 +143,26 @@ class ITensorProxy {
         ttype_(TensorType::kSIMPLE) {}
 
   bool is_trt_tensor() const {
-    assert(validate());
-    assert(ttype_ == TensorType::kTRT);
+    CHECK(validate());
     return trt_tensor_ != nullptr;
   }
 
   bool is_simple_tensor() const {
-    assert(validate());
-    assert(ttype_ == TensorType::kSIMPLE);
+    CHECK(validate());
     return simple_tensor_ != nullptr;
   }
 
   TensorType ttype() const { return ttype_; }
 
   nvinfer1::ITensor* trt_tensor() const {
-    assert(trt_tensor_ != nullptr);
-    assert(ttype_ == TensorType::kTRT);
+    CHECK(trt_tensor_ != nullptr);
+    CHECK(ttype_ == TensorType::kTRT);
```

Comment:
CHECK_EQ probably doesn't work here because of the missing of an output operator for TensorType enum.
We can keep this or add an output operator. I am fine with either way.